app/src/main/java/cz/movapp/app/ui/exercise/play/ExercisePlayAdapter.kt

```diff
@@ -0,0 +1,66 @@
+package cz.movapp.app.ui.exercise.play
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import cz.movapp.app.databinding.ExerciseAudioItemBinding
+import cz.movapp.app.databinding.ExercisePlayItemBinding
+
+
+class ExercisePlayAdapter(
+    private val onAnswerSelected: (ExercisePlayItem, Int) -> AnswerState?,
+    private val onPlayNormal: (ExercisePlayItem) -> Unit,
+    private val onPlaySlow: (ExercisePlayItem) -> Unit,
+) : RecyclerView.Adapter<ViewHolder<ExercisePlayItem>>() {
+
+    companion object {
+        private const val VIEW_TYPE_TEXT = 1
+        private const val VIEW_TYPE_AUDIO = 2
+    }
+
+    private val dataSet: MutableList<ExercisePlayItem> = mutableListOf()
+
+    fun setData(data: List<ExercisePlayItem>) {
+        dataSet.clear()
+        dataSet.addAll(data)
+        notifyDataSetChanged()
```

Comment:
you could use methods like notifyItemChanged(int), notifyItemInserted(int),  notifyItemRemoved(int)
To get away with this you could use https://developer.android.com/reference/androidx/recyclerview/widget/DiffUtil